packages/common/src/common.ts

```diff
@@ -375,8 +401,11 @@ export class Common extends EventEmitter {
    * @param td
    * @returns The name of the HF set
    */
-  setHardforkByBlockNumber(blockNumber: BigIntLike, td?: BigIntLike): string {
-    const hardfork = this.getHardforkByBlockNumber(blockNumber, td)
+  setHardforkByBlockNumber(
+    blockNumber: BigIntLike,
+    { td, timestamp }: { td?: BigIntLike; timestamp?: BigIntLike } = {}
```

Comment:
Doesn't this change of `td` to an object make this a breaking change? I'm assuming it would be preferable to just add an additional optional parameter got timestamp.

Reply:
Yes, changing to an object is a breaking change (also for the `getHardforkByBlockNumber` method), so we would need to switch to an additional parameter.